docs/providers/aws/events/cognito-user-pool.md

```diff
@@ -74,26 +74,76 @@ functions:
           trigger: PreSignUp
 ```
 
-## Using existing pools
+## Special Trigger Considerations
 
-Sometimes you might want to attach Lambda functions to existing Cognito User Pools. In that case you just need to set the `existing` event configuration property to `true`. All the other config parameters can also be used on existing user pools:
+### Custom Sender Triggers
 
-**IMPORTANT:** You can only attach 1 existing Cognito User Pool per function.
+There are two types of Custom Sender Triggers, `CustomSMSSender` and `CustomEmailSender`, both [documented by AWS](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-sender-triggers.html).
 
-**NOTE:** Using the `existing` config will add an additional Lambda function and IAM Role to your stack. The Lambda function backs-up the Custom Cognito User Pool Resource which is used to support existing user pools.
+In order to use these triggers, you must supply a `kmsKeyId` and (optionally) the `lambdaVersion` of the function. Only 1 `kmsKeyId` can be supplied per Cognito User Pool.
 
-```yaml
+```yml
 functions:
-  users:
-    handler: users.handler
+  customSMSSenderFunction:
+    handler: customSMSSender.handler
     events:
       - cognitoUserPool:
-          pool: legacy-user-pool
-          trigger: CustomMessage
-          existing: true
+          pool: MyUserPool1
+          trigger: CustomSMSSender
+          # Required with CustomSMSSender and CustomEmailSender Triggers
+          # Can either be KMS Key ARN string or reference to KMS Key Resource ARN (see customEmailSenderFunction below)
+          kmsKeyId: 'arn:aws:kms:eu-west-1:111111111111:key/12345678-9abc-def0-1234-56789abcdef1'
+          # Optional, defaults to 'V1_0'
+          lambdaVersion: 'V1_0'
```

Comment:
I think there's no point in requiring users to provide this if this is the only possible value - if there's a new one, we can introduce it as an optional field with the default of `V1_0` instead. For now I would just always assume it's `V1_0` - just to clarify, it's mandatory to pass this value to config right now?

Reply:
Curious...

From the AWS docs, the `LambdaVersion` is not a required key, but when omitted, the request fails and AWS says it is a required key. Because of this, I am going to keep the logic the same, but disable the ability to pass in a different lambda version.

If we want to further prune this, we can, but I believe keeping how it is would make it easier to integrate other versions later on.